File: technical/unique_string/golang/unique_string.go
package unique_string

import "sort"

func IsUnique1(s string) bool {
	chars := make(map[rune]bool)
	for _, b := range s {
		if _, ok := chars[b]; ok {
			return false
		}
		chars[b] = true
	}
	return true
}

func IsUnique2(s string) bool {
	for i, a := range s {
		for j, b := range s {
			if i != j && a == b {
				return false
			}
		}
	}
	return true
}

type sortedRunes []rune

func (s sortedRunes) Len() int {
	return len(s)
}

func (s sortedRunes) Less(i, j int) bool {
	return s[i] < s[j]
}

func (s sortedRunes) Swap(i, j int) {
	tmp := s[i]
	s[i] = s[j]
	s[j] = tmp
}

func IsUnique3(s string) bool {
	r := []rune(s) // certain characters in go can be multi-byte, but will be single rune
	sort.Sort(sortedRunes(r))
	for i := 0; i < len(r)-1; i++ {
		if r[i] == r[i+1] {
			return false
		}
	}
	return true
}

File: technical/unique_string/c++/unique_string.cpp
#include <algorithm>

bool isUnique1(char* s) {
  bool map[256];
  for(int i = 0; i < 256; i++) {
    map[i] = false;
  }
  int x = 0;
  while(s[x]) {
    if(map[s[x]]) {
      return false;
    }
    map[s[x]] = true;
    x++;
  }
  return true;
}

bool isUnique2(char* s) {
  int len = 0;
  while(s[len]) len++;
  for(int i = 0; i < len-1; i++) {
    for(int j = i+1; j < len; j++) {
      if(s[i] == s[j]) {
        return false;
      }
    }
  }
  return true;
}

bool isUnique3(char* s) {
  int len = 0;
  while(s[len]) len++;
  if(len < 2) {
    return true;
  }
  std::sort(s, s+len);
  for(int i = 0; i < len-1; i++) {
    if(s[i] == s[i+1]) {
      return false;
    }
  }
  return true;
}